src/main/java/ch/njol/skript/expressions/ExprIndicesOfX.java

```diff
@@ -0,0 +1,173 @@
+)
+@Examples({
+	"set {_list::*} to 1, 2, 3, 1, 2, 3",
+	"set {_indices::*} to the indices of the value 1 in {_list::*}",
+	"# {_indices::*} is now \"1\" and \"4\"",
+	"",
+	"set {_indices::*} to the indices of the value 2 in {_list::*}",
+	"# {_indices::*} is now \"2\" and \"5\"",
+	"",
+	"set {_positions::*} to the positions of the value 3 in {_list::*}",
+	"# {_positions::*} is now 3 and 6",
+	"",
+	"set {_otherlist::bar} to 100",
+	"set {_otherlist::hello} to \"hi\"",
+	"set {_otherlist::burb} to 100",
+	"set {_otherlist::tud} to \"hi\"",
+	"set {_otherlist::foo} to 100",
+	"",
+	"set {_indices::*} to the first index of the value 100 in {_otherlist::*}",
+	"# {_indices::*} is now \"bar\"",
+	"set {_indices::*} to the last index of the value 100 in {_otherlist::*}",
+	"# {_indices::*} is now \"foo\"",
+	"",
+	"set {_positions::*} to the positions of the value 100 in {_otherlist::*}",
+	"# {_positions::*} is now 1, 3 and 5",
+	"set {_positions::*} to the positions of the value \"hi\" in {_otherlist::*}",
+	"# {_positions::*} is now 2 and 4"
+})
+@Since("INSERT VERSION")
+public class ExprIndicesOfX extends SimpleExpression<Object> {
+
+	static {
+		Skript.registerExpression(ExprIndicesOfX.class, Object.class, ExpressionType.COMBINED,
+			"[the] [1:first|2:last] (indices|index[es]) of [[the] value] %object% in %~objects%",
+			"[the] [1:first|2:last] position[s] of [[the] value] %object% in %~objects%"
+		);
+	}
+
+	private IndexType type;
+	private boolean position;
+	private Expression<?> value;
+	private Expression<?> objects;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (exprs[1].isSingle()) {
+			Skript.error("'" + exprs[1] + "' can only ever have one value at most, thus the 'indices of x in list' expression has no effect.");
+			return false;
+		}
+
+		if (!(exprs[1] instanceof Variable<?>) && matchedPattern == 0) {
+			Skript.error("'" + exprs[1] + "' is not a list variable. You can only get the indices of a list variable.");
+			return false;
+		}
+
+		type = IndexType.values()[parseResult.mark];
+		position = matchedPattern == 1;
+		value = LiteralUtils.defendExpression(exprs[0]);
+		objects = exprs[1];
+
+		return LiteralUtils.canInitSafely(value);
+	}
+
+	@Override
+	protected Object @Nullable [] get(Event event) {
+		Object value = this.value.getSingle(event);
+		if (value == null)
+			return (Object[]) Array.newInstance(getReturnType(), 0);
+
+		if (this.position) {
+			List<Integer> positions = new ArrayList<>();
+
+			int position = 1;
+			for (Object object : objects.getArray(event)) {
+				if (object.equals(value)) {
+					if (type == IndexType.FIRST)
+						return new Integer[]{position};
+					positions.add(position);
+				}
+				position++;
+			}
+
+			if (type == IndexType.LAST)
+				return new Integer[]{positions.get(positions.size() - 1)};
+			return positions.toArray();
+		}
+
+		Variable<?> list = (Variable<?>) objects;
```

Comment:
Unsafe, need to check if it's actually a variable

Reply:
It can never be not a variable. Positions is always true when it's a non-variable

Reply:
Oh I see you've got a check up in init
Just throw an assert here then